homeassistant/components/google_assistant/trait.py

```diff
@@ -1351,12 +1351,17 @@ async def execute(self, command, data, params, challenge):
             _verify_pin_challenge(data, self.state, challenge)
             service = SERVICE_ALARM_DISARM
 
+        pin = (
+            data.config.secure_devices_pin[0]
```

Comment:
Secure devices pin when used with the cloud integration can still be a single value. You need to update `homeassistant/components/cloud/google_config.py` to make sure it returns a list.